python/benchmark/gen_data_distributed.py

```diff
@@ -816,10 +831,7 @@ def make_sparse_regression_udf(
                     sparse_matrix = sparse_matrix[:, informative_shuffle_indices]
                 informative = sparse_matrix[:, :n_informative]
 
-                if use_cupy:
-                    redundant_mul = np.random.rand(n_informative, redundant_cols)
```

Comment:
Was this a bug to use np vs cp here?

Reply:
Not a bug. redundant_mul must be a numpy array because it is used to construct a CPU-based SciPy sparse matrix, regardless of whether use_cupy is true or false. cupy is only utilized for generating the label column.

The cupyx sparse matrix is not used here, which may be fine for now.